src/Form/Type/MissionTypeAssetTypesType.php

```diff
@@ -4,20 +4,36 @@
 
 namespace App\Form\Type;
 
+use App\Entity\Organization;
+use App\Repository\AssetTypeRepository;
 use Symfony\Component\Form\AbstractType;
 use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
 use Symfony\Component\Form\Extension\Core\Type\IntegerType;
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Security\Core\Security;
 
 class MissionTypeAssetTypesType extends AbstractType
 {
+    protected array $assetTypes;
+
+    public function __construct(Security $security, AssetTypeRepository $assetTypeRepository)
+    {
+        /** @var Organization $organization */
+        $organization = $security->getUser();
+
+        $assetTypes = $assetTypeRepository->findByOrganization($organization->getParentOrganization());
+        foreach ($assetTypes as $assetType) {
+            $this->assetTypes[$assetType->id] = $assetType->name;
+        }
+    }
+
     public function buildForm(FormBuilderInterface $builder, array $options): void
     {
         $builder
             ->add('type', ChoiceType::class, [
                 'label' => 'Type',
-                'choices' => CommissionableAssetType::TYPES,
+                'choices' => array_flip($this->assetTypes),
```

Comment:
wouldn't an `EntityType` + `query_builder` would be cleaner?

Reply:
This was done on purpose, to avoid the full serialized object to be stored in the JSON field. 
I've added a comment in the code